### PR TITLE
Use impersonated token to prevent user token expiration

### PR DESCRIPTION
**Pull request by ewelinajozwik**
Issue: [TMT-1047](https://jira.thehyve.nl/browse/TMT-1047)
Related UI change: https://github.com/thehyve/Fractal.js/pull/1

Further authorization improvements: [TMT-654](https://jira.thehyve.nl/browse/TMT-654), see also https://github.com/LCSB-BioCore/Fractalis/pull/3

### CODE REVIEW

**Comment by ewelinajozwik** [comment]
Still in progress:
- [x] fix tests - test build freezes after test collect
- [x] update documentation - describe new configuration parameters in README

File: fractalis/authorization.py [hunk 1]
```diff
@@ -9,38 +9,47 @@
...
+
+    def authorize():
+        token = get_request_token()
+        if not token or len(token) == 0:
+            error = "No token in the authentication object."
+            logger.error(error)
+            raise Unauthorized(error)
+        decoded_token = jwt.decode(token, verify=False)
+        client_id = app.config.get('OIDC_CLIENT_ID')
+        url = app.config.get('OIDC_SERVAUTH_SERVICE_TYPEER_URL')
```
Comment (gijskant) [inline fractalis/authorization.py]: Copy-paste error?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,19 @@ Tip: Use the [default settings](fractalis/config.py) as an example for your own
...
 Please don't overwrite default settings if you don't know what you are doing. This might have severe implications for security or might cause Fractalis to not work correctly.
 
+
+#### Authorization
+
+Configuration options for authorization, currently specific to the transmart handler:
+
+| Configuration          | Default value | Description                                                                                             |
+|------------------------|---------------|---------------------------------------------------------------------------------------------------------|
+| AUTHORIZATION_DISABLED | False         | Disable validation of an access token from a request. Disabling is recommended!                                                  |
```
Comment (gijskant) [inline README.md]: Shouldn't this be `not recommended`?

Comment (ewelinajozwik) [reply README.md]: Oops, I deleted too many letters...

**Review by gijskant — APPROVED** [review]
Looks good. I will also try to test it with glowing bear.

File: fractalis/config.py [hunk 3]
```diff
@@ -35,4 +35,12 @@
 # Whether to verify the certs of https data sources
 ETL_VERIFY_SSL_CERT = False
 
+# Authorization
+# Disable the authorization based on the user access token. Not recommended!
+AUTHORIZATION_DISABLED = False
+# OIDC refresh token enabling an offline access
+OIDC_OFFLINE_TOKEN = '<your_offline_token>'
```
Comment (gijskant) [inline fractalis/config.py]: One comment, these are default values... Shouldn't these be `None`, to make sure they are always overwritten (and fail somewhere when no value has been set?)

Comment (ewelinajozwik) [reply fractalis/config.py]: good point!

**Pull request merged**